category_encoders/quantile_encoder.py

```diff
@@ -0,0 +1,291 @@
+        self.verbose = verbose
+        self.cols = cols
+        self.ordinal_encoder = None
+        self._dim = None
+        self.mapping = None
+        self.handle_unknown = handle_unknown
+        self.handle_missing = handle_missing
+        self.feature_names = None
+        self.quantile = quantile
+        self.m = m
+
+    def fit(self, X, y, **kwargs):
+        """Fit encoder according to X and y.
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+        Returns
+        -------
+        self : encoder
+            Returns self.
+        """
+
+        # unite the input into pandas types
+        X = util.convert_input(X)
+        y = util.convert_input_vector(y, X.index)
+
+        if X.shape[0] != y.shape[0]:
+            raise ValueError(
+                "The length of X is "
+                + str(X.shape[0])
+                + " but length of y is "
+                + str(y.shape[0])
+                + "."
+            )
+
+        self._dim = X.shape[1]
+
+        # if columns aren't passed, just use every string column
+        if self.cols is None:
+            self.cols = util.get_obj_cols(X)
+        else:
+            self.cols = util.convert_cols_to_list(self.cols)
+
+        if self.handle_missing == "error":
+            if X[self.cols].isnull().any().any():
+                raise ValueError("Columns to be encoded can not contain null")
+
+        self.ordinal_encoder = OrdinalEncoder(
+            verbose=self.verbose,
+            cols=self.cols,
+            handle_unknown="value",
+            handle_missing="value",
+        )
+        self.ordinal_encoder = self.ordinal_encoder.fit(X)
+        X_ordinal = self.ordinal_encoder.transform(X)
+        self.mapping = self.fit_quantile_encoding(X_ordinal, y)
+
+        X_temp = self.transform(X, override_return_df=True)
+        self.feature_names = list(X_temp.columns)
+
+        if self.drop_invariant:
+            self.drop_cols = []
+            X_temp = self.transform(X)
+            generated_cols = util.get_generated_cols(X, X_temp, self.cols)
+            self.drop_cols = [x for x in generated_cols if X_temp[x].var() <= 10e-5]
+            try:
+                [self.feature_names.remove(x) for x in self.drop_cols]
+            except KeyError as e:
+                if self.verbose > 0:
+                    print(
+                        "Could not remove column from feature names."
+                        "Not found in generated cols.\n{}".format(e)
+                    )
+
+        return self
+
+    def fit_quantile_encoding(self, X, y):
+        mapping = {}
+
+        # Calculate global statistics
+        prior = self._quantile = np.quantile(y, self.quantile)
+        self._sum = y.sum()
```

Comment:
you actually don't use neither `self._sum` nor `self._count`, do you? So I think we can just drop these two lines. Also the `self._quantile` does not need to be defined